change check to array, fix existing themes

adiux opened the pull request:
#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Check the list of available themes. Among others "Blank" should be available. These themes should **not** be available:
  - Brienz
  - Confirm Me
  - Paprika
3. Enable grapesjs
4. The 3 following 3 themes should now be available in addition to "Blank". 
  - Brienz
  - Confirm Me
  - Paprika

fixes: https://github.com/mautic/plugin-grapesjs-builder/issues/51

RaphaelWoude committed:
Updated dependencies of 7.4 to latest minor versions.

dennisameling committed:
Merge branch 'features' into php8.0

RaphaelWoude committed:
Updated almost all dependencies to support php 8.0

RaphaelWoude committed:
Updated imports for object manager

RaphaelWoude committed:
Updated fos_rest config using the 3.0 upgrade guide.

RaphaelWoude committed:
Changed jms_serializer.metadata.cache from 'none' to 'file'

RaphaelWoude committed:
Removed artifacts of doctrines cache bundle in favor of symfony cache.

RaphaelWoude committed:
Updated ext dependencies to *

RaphaelWoude committed:
Updated oneup/uploader-bundle from 2.1.5 to 3.1.0

dennisameling committed:
Fix OAuthServerBundle

dennisameling committed:
Revert KNP Menu bundle to v2

dennisameling committed:
Fix post-install and post-update scripts 🚀

dennisameling committed:
Don't fail fast for now

dennisameling committed:
Refactor mautic.http.connector (joomla/http) to mautic.http.client (Guzzle)

Commit:
change check to array, fix existing themes

Commit:
change check to array, fix existing themes

Commit:
use the same blank theme for grapesjs and legacy

Commit:
use the same blank theme for grapesjs and legacy

RCheesley commented:
@adiux looks like we have [three failing tests](https://github.com/mautic/mautic/pull/9841/checks?check_run_id=2212002847#step:9:65) to address - I just re-ran the actions to be sure that it wasn't a hiccup with the pipeline.

dennisameling committed:
Fix $coreIntegration types in OneSignal and Twilio integrations

dennisameling committed:
Merge remote-tracking branch 'upstream/features' into php8.0

dennisameling committed:
Make the willdurand/oauth-server-bundle work with Symfony 4 🎉

dennisameling committed:
Upgrade KNP Menu to v3

dennisameling committed:
Remove PHP 8.0 from CI matrix again

dennisameling committed:
friendsofsymfony/oauth-server-bundle back to 1.6 for now

dennisameling committed:
Fix PHPSTAN

dennisameling committed:
Bump PHPSTAN to latest version 🚀

dennisameling committed:
Update Rector to 0.9.31, migrate config to PHP

dennisameling committed:
Use temp friendsofsymfony/oauth-server-bundle version for Doctrine upgrade

dennisameling committed:
CS fixer back to 2.16.1 for now

dennisameling committed:
Make CacheStorageHelper nullable

dennisameling committed:
Temp remove Rector from CI

dennisameling committed:
Re-enable fail-fast in CI

dennisameling committed:
Change rector.yaml to rector.php in excluded files

dennisameling committed:
Remove old sensio/distribution-bundle script from release script

dennisameling committed:
Update upgrade docs

dennisameling committed:
Fix ConsoleErrorListener

dennisameling committed:
Fix MAUTIC_ENV definition

fedys committed:
Shell verbosity restored (#9853)

mabumusa1 committed:
remove hardcoded ids within fixtures (#9846)

mabumusa1 committed:
change auditLog to auditlog (#9845)

mabumusa1 committed:
changes to encoder to update to Symfony (#9842)

* changes to encoder to update to Symfony

* fix to install service

* fixes to encodepassword method

mabumusa1 committed:
fix monolog to use mautic logger (#9840)

alanhartless committed:
TPROD-143 Replaces Froala with CKEditor 4 as the default WYSIWYG (#9843)

* Add ckeditor

* Add token tag feature

* install new plugin

* implement token

* Change visibility of function

* fix css issue

* Recommendation changes in code review

* Recommendation changes in code review

* Add version for asset URLs

* functional test cases

* Elfinder

* Replace froala editor to ckeditor in dynamic content

* Add font families

* Code review recommended changes

* tearDown() is final now

* Fix undefined method error

* We have to convert fonts to a format that is supported by CK Editor.

* downgrade ckeditor 5 to 4

* Fix test case

* Fix conflict

* removed unused files

* Load editor font URLs

* Reset prod app.css to features to see if it won't cause conflicts in the PR

* Copy changes made directly to CoreBundle/Assets/css/app.css to applicable less files

* Fixed failing test

Co-authored-by: saurabh.gupta <saurabh.gupta@acquia.com>
Co-authored-by: fedys <miroslav.fedeles@gmail.com>
Co-authored-by: Anton Vlasenko <vlasenko.anton@gmail.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

kuzmany committed:
Integrations Bundle - add support to pass custom options to OrderDAO (#9796)

* Integrations Bundle - add support to pass custom options to OrderDAO

* Options empty by default

* Stricter typing for arrays

Co-authored-by: Dennis Ameling <dennis@dennisameling.com>

kuzmany committed:
Clickable campaign ID in contact's campaigns list (#9666)

Co-authored-by: Dennis Ameling <dennis@dennisameling.com>

kuzmany committed:
Allow change default theme for new email (#9189)

* Allow change default theme for new email

* Sricter types

Co-authored-by: Dennis Ameling <dennis@dennisameling.com>

alanhartless committed:
Aggregate stat service (#9757)

* StatBundle for stat collection

* Stat DAO that calculate

* Dispatch event to collect the stats

* Created an object to store stats

* Calculator for summing and averaging stats

* Fixed stat objects where getStat was not useful.

* Add StatsDAO test

* Added proxy to get Calculator

* Collector test

* Format key once so it doesn't have to be done everywhere else

* Fixed average methods

* Add means to pass options to listeners

* removed get average in favor of get count

* Fixed namespace

* Add config file

* Resolve conflicts with c6acef3d10e50ab0063816d161dbcc2123b40269

* Resolve conflicts with 36df6286210943245b40e15f94b94eb945b854e9

* Fixed filling in missing days and got rid of unused average calculator methods for now

* Resolve conflicts with 9d5024ba335b8145b2014fef2eb7d34ef3356323

* Resolve conflicts with c1f7b5111238dd57dece26ef25faa1e8b6a035b2

* Resolve conflicts with 8ba2b291f0f1c0a411cfdc97f35cc47c398af11a

* Fixed bounced/unsubscribed queries

* return values not keys for graphs

* Resolve conflicts with 56a394e3fa843aa7c902bdb53e04a1e546a437aa

* Removed undefined method use

* Resolve conflicts with 8f2f70dff2d65834d61481a531934af7e33e1fc1

* fix data added by year only, fix competing missing data in set, add sum method to calculator

* fix merge problems

* Resolve conflicts with cf133a96cede4db359d3b36396776eacde3ad661

* fix invalid argument

* make dashboard cache to CacheBundle, refactor event to use the cache bundle, replace event constuctions with factory calls,

* remove strict declarations

* Update app/bundles/StatsBundle/Event/AggregateStatRequestEvent.php

Co-Authored-By: galvani <galvani78@gmail.com>

* Update app/bundles/StatsBundle/Event/AggregateStatRequestEvent.php

Co-Authored-By: galvani <galvani78@gmail.com>

* Update app/bundles/StatsBundle/Event/AggregateStatRequestEvent.php

Co-Authored-By: galvani <galvani78@gmail.com>

* Update app/bundles/StatsBundle/Event/AggregateStatRequestEvent.php

Co-Authored-By: galvani <galvani78@gmail.com>

* Resolve conflicts with 95c5baad70e90cf4904bb2d93adcd8f73c8dcd33

* Fixed issue in graphs that didn't take into account the given email IDs so all stats were displayed.

* Resolved conflicts with 899ecd237b884f629d4f51631f33870a5ab700b8

* Add tests

* Fix docblock

* Add test

* Others than creator can view stats

* Revert "Others than creator can view stats"

This reverts commit acfdbeae9b22dcb1150b9bdc17ccb5af8ba60868.

* Resolving conflicts with ea278cc3077531d78f519ee9324a470dd5a20a91

* Resolved conflicts with d52c77f9342788655e8c2904631420cbf864c7ef

* Resolved conflicts with ffd1d01c28327938f233dc4e84964164a0dce822

* Fixed bad conflict resolution

* Fixed bad conflict resolution

* handle empty string as null filterTrait

* Fixed bad merge

* Use generated column if available to speed up query from 15 minutes to 1 second

* Use 1 method to generate units from date range. It must return the same result.

* CS fix

* DateRangeUnitTrait fully tested

* add week methods to stats, fix addStat to use 24hrs format, create week format methods

* fix code style

* revert stupidity

* Update app/bundles/StatsBundle/Aggregate/Collection/Stats/WeekStat.php

update phpdoc year

Co-Authored-By: galvani <galvani78@gmail.com>

* Update app/bundles/StatsBundle/Aggregate/Helper/CalculatorHelper.php

Co-Authored-By: galvani <galvani78@gmail.com>

* Update app/bundles/StatsBundle/Aggregate/Helper/CalculatorHelper.php

Co-Authored-By: galvani <galvani78@gmail.com>

* Update app/bundles/StatsBundle/Aggregate/Helper/CalculatorHelper.php

Co-Authored-By: galvani <galvani78@gmail.com>

* make correct use of week unit in queries

* remove comment

* simplify week data calculation

* Fixed tests

* Delete DateRangeUnitTraitTest.php

Deleting file in "unit" to be re-added under "Unit" (aggravating case insensitive filesystem)

* Added back DateRangeUnitTraitTest

* Fixed namespace for DateRangeUnitTraitTest

* CS fixes

* Rector fixes

* PHPStan fixes from Rector fixes

* More PHPStan fixes from Rector fixes

Co-authored-by: Jan Kozak <galvani78@gmail.com>
Co-authored-by: John Linhart <jan@linhart.email>
Co-authored-by: John Linhart <admin@escope.cz>
Co-authored-by: Lukas Drahy <lukas@drahy.net>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

kuzmany committed:
Add search by company fields for list view/API (#8352)

* Add search by company field for list view/API

* Fix travis issues

* Fix travis issue

* Add wildcard to search

Co-authored-by: Dennis Ameling <dennis@dennisameling.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

mabumusa1 committed:
Check if form calls IsSubmitted (#9847)

* Check if form IsSubmitted

* fix behaviour of isValid

* Update app/bundles/CoreBundle/Controller/AbstractFormController.php

Co-authored-by: John Linhart <jan@linhart.email>

Co-authored-by: John Linhart <jan@linhart.email>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

escopecz committed:
A/B test chart rendering unified (#7787)

* Use one template to render all charts

* Fixing empty space caused by forgotton canvas in the HTML

Co-authored-by: Norman Pracht - Webmecanik <npr@webmecanik.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>
Co-authored-by: Dennis Ameling <dennis@dennisameling.com>

dennisameling committed:
Use official Codecov GH Action

escopecz committed:
Marketplace v1 (#9650)

* Read-only Marketplace UI + list command

* Show the confirm button only if some confirm text provided

* Register the MarketplaceBundle in AppKernel

* View and create(install) permissions implemented

* Feature flag marketplace_enabled implemented

* ListCommand tested via Functional test

* Functional test for the marketplace list table

* PHP 7.4 arrow functions

* Container fix

* PHP 7.4 typed properties

* Replacing deprecated interface

* Basic detail functional test added

* Composition over inheritance

* Detail page redesign

* Updated link to the Marketplace user docs

* PHPSTAN fixes

* CS fixes

* Typo fix

* Adding undefined property (PHPSTAN found it)

* Removing unused code to improve coverage

* Adding a link to the packagist detail page to improve coverage and UX

* Making value objects final as there is no need to mock or extend them

* Replacing getters, setters with typed public properties for Version DTO

* Replacing getters, setters with typed public properties for PackageBase DTO

* Replacing getters, setters with typed public properties for Maintainer DTO

* Replacing getters, setters with typed public properties for PackageDetail DTO

* Replacing getters, setters with typed public properties for GitHubInfo DTO

dennisameling committed:
Verbose logs are a bit too much

kuzmany committed:
New report: Contact's segments activity log (#8125)

* New report: Contact's segments activity log

* Remove CommonSubscriber

* New report: Contact's segments activity log

* Remove CommonSubscriber

* Allow group by for segments activity log report

* Add SegmentLogReportSubscriberTest.php

Co-authored-by: Norman Pracht - Webmecanik <npr@webmecanik.com>
Co-authored-by: Dennis Ameling <dennis@dennisameling.com>
Co-authored-by: Julien Ulm <jul@webmecanik.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

dennisameling committed:
Merge branch 'features' into codecov-gh-actions

dongilbert committed:
[WIP] TPROD-170: Enabling 2-legged authentication for Mautic (#9837)

* Enabling 2-legged authentication for Mautic

Add prefix to table name in foreign key constraint

Adding unit test

Fixing namespace issue in test class

Fixing migration issue and adding AuditLog username

Making user_id field nullable for client_credentials grant type

Cascade persist role in oauth_client

Fixing role_id issue in oauth2_clients table

Tweaked the audit log name and captured first/last name for "created by metadata"

* MAUT-4344: Check oauth2_accesstokens table's user_id field type before running migration

Co-authored-by: Liju P M <liju.pm@acquia.com>
Co-authored-by: Ruth Cheesley <ruth.cheesley@acquia.com>

RCheesley committed:
Merge pull request #9833 from RaphaelWoude/php8.0

Upgrade dependencies (BC breaks) in preparation for PHP 8.0 support

RCheesley committed:
Merge pull request #9850 from dennisameling/fix-console-error-listener

Fix ConsoleErrorListener

dennisameling committed:
Merge branch 'features' into codecov-gh-actions

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: create .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: create .all-contributorsrc [skip ci]

RCheesley committed:
Merge pull request #9858 from mautic/all-contributors/add-dennisameling

docs: add dennisameling as a contributor

RCheesley committed:
Merge branch 'features' into all-contributors/add-rcheesley

RCheesley committed:
Merge pull request #9859 from mautic/all-contributors/add-rcheesley

docs: add rcheesley as a contributor

RCheesley committed:
Update .all-contributorsrc

RCheesley committed:
Update .all-contributorsrc

RCheesley committed:
Update .all-contributorsrc

RCheesley committed:
Update .all-contributorsrc

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

RCheesley committed:
Merge pull request #9861 from mautic/all-contributors/add-mabumusa1

docs: add mabumusa1 as a contributor

RCheesley committed:
Merge pull request #9856 from dennisameling/codecov-gh-actions

Use official Codecov GH Action

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

RCheesley committed:
Merge pull request #9862 from mautic/all-contributors/add-escopecz

docs: add escopecz as a contributor

nickveenhof committed:
Allowing composer based installations and move plugins & themes to separate repositories. Also copying asset files to app/assets but keep duplicates in there for now

nickveenhof committed:
Updating dependencies

nickveenhof committed:
Updating dependencies

nickveenhof committed:
Removing plugins from the codebase

nickveenhof committed:
Updating dependencies

nickveenhof committed:
Code standards fix

nickveenhof committed:
Updating to latest core-lib

nickveenhof committed:
Updating composer file and the correct htaccess files in the correct folders

alanhartless committed:
Re-enable the fos_rest.body_listener to parse json requests

alanhartless committed:
getModel must use point.triggerevent instead of point.triggerEvent because the container is now case sensitive and TriggerEventModel is registered as mautic.point.model.triggerevent.

alanhartless committed:
DeviceDetector __construct($userAgent) is now type casted with string which the API test that hits the tracking pixel does not set resulting in a 500.

alanhartless committed:
Empty choices must be added as `placeholder` to pass validation

RCheesley committed:
Merge pull request #9865 from alanhartless/m4-alpha-api-fixes

M4 API fixes

RCheesley committed:
Merge branch 'features' into new-composer-flow

nickveenhof committed:
Merge branch 'features' of github.com:mautic/mautic into new-composer-flow

nickveenhof committed:
Updating with new mautic/core

nickveenhof committed:
Merge branch 'new-composer-flow' of github.com:nickveenhof/mautic into new-composer-flow

nickveenhof committed:
Fix PHPSTAN issues

nickveenhof committed:
Updating dependencies

nickveenhof committed:
Updating dependencies

nickveenhof committed:
Downgrade framework temporarily as this breaks mautic unit testing

alanhartless committed:
Removed unnecessary call to execute that caused an exception on dbal 2.13.0

alanhartless committed:
Fixed test compatibility with Symfony Framework 4.4.21

alanhartless committed:
Fixed test compatibility with Symfony Framework 4.4.21

alanhartless committed:
Removed use of deprecated at()

alanhartless committed:
Merge pull request #9867 from nickveenhof/update-dependencies

Patch and minor version bumps for some dependencies

nickveenhof committed:
Updating tests and dependencies

nickveenhof committed:
Resolving composer lock conflicts

dennisameling committed:
Add .htaccess to app/assets folder

RCheesley committed:
Merge pull request #9863 from nickveenhof/new-composer-flow

composerisation of Mautic TPROD-147

dennisameling committed:
Allow manually generating release notes

RCheesley committed:
Merge pull request #9868 from dennisameling/release-notes

[skip ci] Allow manually generating release notes

Commit:
fix: add blank to legacy builder, improve check for empty template

Commit:
fix: add blank to legacy builder, improve check for empty template

Commit:
fix: testcase, improve error message

Commit:
fix: testcase, improve error message

Commit:
cleanup after merge conflicts: remove config.json from deleted confimme

Commit:
Merge branch 'feature/allow-theme-to-be-used-with-mulitple-builders' of https://github.com/adiux/mautic into feature/allow-theme-to-be-used-with-mulitple-builders

# Conflicts:
#	themes/confimme/config.json
